helper/helper.go

```diff
@@ -138,33 +139,31 @@ func InitializeGlobals() {
 func InitializeSignalHandler() {
 	signalChan := make(chan os.Signal, 1)
 	signal.Notify(signalChan, unix.SIGINT, unix.SIGTERM, unix.SIGPIPE, unix.SIGUSR1)
-	terminatedChan := make(chan bool, 1)
 	for {
 		go func() {
 			sig := <-signalChan
 			fmt.Println() // Add newline after "^C" is printed
 			switch sig {
 			case unix.SIGINT:
 				gplog.Warn("Received an interrupt signal on segment %d: aborting", *content)
-				terminatedChan <- true
+				wasTerminated.Store(true)
 			case unix.SIGTERM:
 				gplog.Warn("Received a termination signal on segment %d: aborting", *content)
-				terminatedChan <- true
+				wasTerminated.Store(true)
 			case unix.SIGPIPE:
 				if *onErrorContinue {
 					gplog.Warn("Received a broken pipe signal on segment %d: on-error-continue set, continuing", *content)
-					terminatedChan <- false
+					wasTerminated.Store(false)
 				} else {
 					gplog.Warn("Received a broken pipe signal on segment %d: aborting", *content)
-					terminatedChan <- true
+					wasTerminated.Store(true)
 				}
 			case unix.SIGUSR1:
 				gplog.Warn("Received shutdown request on segment %d: beginning cleanup", *content)
-				terminatedChan <- true
+				wasTerminated.Store(true)
 			}
 		}()
-		wasTerminated = <-terminatedChan
-		if wasTerminated {
+		if wasTerminated.Load() {
```

Comment:
Wait on a channel is lost here and it looks like busy loop now. I suggest revert to channel here and store value read from the channel.

Reply:
done